app/src/main/java/com/infomaniak/mail/data/cache/mailboxContent/DraftController.kt

```diff
@@ -87,36 +89,49 @@ object DraftController {
     }
 
-    fun MutableRealm.setDraftSignature(draftLocalUuid: String) {
-        updateDraft(draftLocalUuid, this) { draft ->
+    fun MutableRealm.setSignature(draft: Draft) {
 
-            draft.mimeType = ClipDescription.MIMETYPE_TEXT_HTML
+        draft.mimeType = ClipDescription.MIMETYPE_TEXT_HTML
 
-            val defaultSignature = SignatureController.getDefaultSignature(this) ?: return@updateDraft
+        SignatureController.getDefaultSignature(this)?.let { defaultSignature ->
 
-            if (draft.identityId == null) draft.identityId = defaultSignature.id
+            draft.identityId = defaultSignature.id
 
-            if (draft.from.isEmpty()) {
-                draft.from = realmListOf(Recipient().apply {
-                    this.email = defaultSignature.sender
-                    this.name = defaultSignature.fullName
-                })
+            draft.from = realmListOf(Recipient().apply {
+                this.email = defaultSignature.sender
+                this.name = defaultSignature.fullName
+            })
+
+            draft.replyTo = realmListOf(Recipient().apply {
+                this.email = defaultSignature.replyTo
+                this.name = ""
+            })
+
+            val html = "<br/><br/><div class=\"editorUserSignature\">${defaultSignature.content}</div>"
+            draft.body = when (defaultSignature.position) {
+                SignaturePosition.AFTER_REPLY_MESSAGE -> draft.body + html
+                else -> html + draft.body
             }
+        }
+    }
 
-            if (draft.replyTo.isEmpty()) {
-                draft.replyTo = realmListOf(Recipient().apply {
-                    this.email = defaultSignature.replyTo
-                    this.name = ""
-                })
+    fun MutableRealm.setPreviousMessage(draft: Draft, draftMode: DraftMode, previousMessageUid: String?) {
+        previousMessageUid?.let { MessageController.getMessage(it, this) }?.let { previousMessage ->
+
+            previousMessage.msgId.let {
+                draft.inReplyTo = it
+                draft.references = it
             }
 
-            if (draft.body.isEmpty()) {
-                val html = "<br/><br/><div class=\"editorUserSignature\">${defaultSignature.content}</div>"
-                draft.body = when (defaultSignature.position) {
-                    SignaturePosition.AFTER_REPLY_MESSAGE -> draft.body + html
-                    else -> html + draft.body
-                }
+            when (draftMode) {
+                DraftMode.REPLY, DraftMode.REPLY_ALL -> draft.inReplyToUid = previousMessageUid
+                DraftMode.FORWARD -> draft.forwardedUid = previousMessageUid
+                DraftMode.NEW_MAIL -> Unit
             }
+
+            draft.to = previousMessage.from
+            if (draftMode == DraftMode.REPLY_ALL) draft.cc = previousMessage.to.union(previousMessage.cc).toRealmList()
+            previousMessage.subject?.let { draft.subject = "Re: $it" }
         }
```

Comment:
All this code is executed if `previousMessageUid` is not null
So we might as well put it directly as non-null in the method to simplify everything

Reply:
Done.